src/prototype_room_my.js

```diff
@@ -357,8 +357,28 @@ Room.prototype.handleReputation = function() {
   }
 };
 
+/**
+ * todo move this to prototype_room_resources.js ?
+ *
+ * @return {number}
+ */
+Room.prototype.getEnergy = function() {
+  let amount = 0;
+  if (this.terminal) {
+    amount += this.terminal.store[RESOURCE_ENERGY];
+  }
+  if (this.storage) {
+    amount += this.storage.store[RESOURCE_ENERGY];
+  }
+  return amount;
+};
+
 Room.prototype.getBuilderAmount = function(constructionSites) {
   let amount = 1;
+  const energyInRoom = this.getEnergy();
+  if (energyInRoom > (config.terminal.minEnergyAmount / 2)) {
```

Comment:
:-) What does this mean?

Reply:
there should be more Builder if we have energy in terminal and are upgrating to RCL 6/7/8, there are a lot of buildings that needs to be done asap (e.g links near 2nd sourcer)

Reply:
This is tricky to understand from that line.
Wouldn't it be better to use `this.isHealthy()`? And maybe extend this. This is reusable and makes clearer what the intention is.